blog/03-15-2023-Sage-Sage-A-Solidity-Smart-Contract-for-Auctioning-Flowers-on-the-celo-Blockchain/Sage-Sage-A-Solidity-Smart-Contract-for-Auctioning-Flowers-on-the-celo-Blockchain.md

```diff
@@ -0,0 +1,462 @@
+        bool sold;
+    }
+event AuctionEnded(
+    address winner,
+    address highestBidder,
+    uint highestBid
+);
+```
+
+ The FloralAuction contract contains a data structure Flower which represents a flower that can be put up for auction. The Flower structure has several properties such as the current owner, the flower's name, description, image, initial price, current highest bid, highest bidder, and flags for indicating if the flower is for sale or has been sold.
+
+The event AuctionEnded is an event that is emitted when the auction for a flower ends. It contains three parameters: winner which represents the address of the winner of the auction, highestBidder which represents the address of the highest bidder, and highestBid which represents the highest bid made during the auction. The event is useful for external applications that want to be notified when an auction ends, for example, to update the user interface with the latest information about the auction.
+
+Additionally, to interact with the cUSD token on the Celo Alfajores test network, it is necessary to include the address of the token.
+
+```solidity
+address internal ownerAddress = 0xb7BF999D966F287Cd6A1541045999aD5f538D3c6;
+ address internal cUsdTokenAddress = 0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1;
+```
+
+
+Next, we add out mapping
+ ```solidity
+  mapping (uint => Flower) internal flowers;
+    uint internal flowersLength = 0;
+    uint public auctionEndTime;
+
+    constructor() {
+        auctionEndTime = block.timestamp + 24 hours;
+    }
+ ```
+  This code initializes a mapping called flowers that maps an unsigned integer (uint) to a Flower struct. The Flower struct contains various details about a flower, such as its owner, name, description, image, initial price, highest bid, highest bidder, and whether it is for sale or sold.
+
+Additionally, the code initializes two unsigned integers: flowersLength and auctionEndTime. flowersLength is set to 0 by default, while auctionEndTime is set to the current block timestamp plus 24 hours.
+
+The constructor function is a special function that is executed only once, when the contract is deployed. In this case, the constructor sets the auctionEndTime to 24 hours after the current block timestamp. This means that the auction will end exactly 24 hours after the contract is deployed.
+ 
+Furthermore, to make our smart contract more interesting we begin to add functions. The first function we will be adding is the `createFlower` function.
+
+```solidity
+    function createFlower(
+        string memory _name,
+        string memory _description,
+        string memory _image,
+        uint _price,
+        bool _isSale
+    ) public {
+        flowers[flowersLength] = Flower(
+            payable(msg.sender),
+            _name,
+            _description,
+            _image,
+            _price,
+            0,
+            address(0),
+            _isSale,
+            false
+        );
+        flowersLength++;
+    }
+
```

Comment:
line (157 and 158) remove the two extra spacing at the end of the code. do the same fixes in other place where it occurs.

Reply:
done